Silicon/RaptorlakePkg/Library/FusaCrashLogLib/FusaCrashLogLib.c

```diff
@@ -0,0 +1,381 @@
+
+  CrashLogAddr = (UINT32 *) (UINTN) MainLogDataBuffer;
+  DEBUG ((DEBUG_VERBOSE, "Value at MainCrashLogAddress 0x%x = 0x%x \n ", (UINTN)CrashLogAddr, (UINT32) *CrashLogAddr));
+  while (CrashLogSize < (MainLogSize)) {
+    *MainLogDestination = *CrashLogAddr;
+    if (CrashLogSize < CRASHLOG_SIZE_DEBUG_PURPOSE) { // Dumping only few bytes to help debug
+      DEBUG ((DEBUG_VERBOSE, "Main CrashData = 0x%x\n", *MainLogDestination));
+    }
+    MainLogDestination++;
+    CrashLogAddr++;
+    CrashLogSize+=4;
+  }
+  DEBUG ((DEBUG_VERBOSE, "Copied Main CrashLog Size = 0x%x\n", CrashLogSize));
+
+  CrashLogSize = 0;
+  CrashLogAddr = (UINT32 *) (UINTN) TelemetryDataBuffer;
+  DEBUG ((DEBUG_VERBOSE, "Value at TelemetryCrashLogAddress 0x%x = 0x%x \n ", (UINTN)CrashLogAddr, (UINT32) *CrashLogAddr));
+  while (CrashLogSize < (TelemetrySize)) {
+    *TelemetryDestination = *CrashLogAddr;
+    if (CrashLogSize < CRASHLOG_SIZE_DEBUG_PURPOSE) { // Dumping only few bytes to help debug
+      DEBUG ((DEBUG_VERBOSE, "Telemetry CrashData = 0x%x\n", *TelemetryDestination));
+    }
+    TelemetryDestination++;
+    CrashLogAddr++;
+    CrashLogSize+=4;
+  }
+  DEBUG ((DEBUG_VERBOSE, "Copied Telemetry CrashLog Size = 0x%x\n", CrashLogSize));
+
+  CrashLogSize = 0;
+  CrashLogAddr = (UINT32 *) (UINTN) TraceDataBuffer;
+  DEBUG ((DEBUG_VERBOSE, "Value at TraceCrashLogAddress 0x%x = 0x%x \n ", (UINTN)CrashLogAddr, (UINT32) *CrashLogAddr));
+  while (CrashLogSize < (TraceSize)) {
+    *TraceDestination = *CrashLogAddr;
+    if (CrashLogSize < CRASHLOG_SIZE_DEBUG_PURPOSE) { // Dumping only few bytes to help debug
+      DEBUG ((DEBUG_VERBOSE, "Trace CrashData = 0x%x\n", *TraceDestination));
+    }
+    TraceDestination++;
+    CrashLogAddr++;
+    CrashLogSize+=4;
+  }
+  DEBUG ((DEBUG_VERBOSE, "Copied Trace CrashLog Size = 0x%x\n", CrashLogSize));
+
+  return EFI_SUCCESS;
+}
+
+/**
+
+  Add a new Firmware CrashLog Entry to existing Error Status Block.
+
+  @param[in] ErrStsBlk                          BERT GENERIC_ERROR_STATUS instance.
+  @param[in out] FirmwareGenericErrorAddr       Pointer to this FirmwareGenericError entry address, updated on return.
+  @param[in out] FirmwareCrashLogPayloadAddr    Pointer to Firmware CrashLog Entry payload address, updated on return.
+  @param[in] EntrySize                          Firmware CrashLog Entry payload size.
+  @param[in] EntrySource                        Firmware CrashLog Entry source.
+  @param[in] EntryVersion                       Firmware CrashLog Entry version.
+
+**/
+
+VOID
+AddFirmwareCrashLogEntry (
+  IN     EFI_ACPI_6_4_GENERIC_ERROR_STATUS_STRUCTURE    *ErrStsBlk,
+  IN OUT FIRMWARE_GENERIC_ERROR                         **FirmwareGenericErrorAddr,
+  IN OUT UINT8                                          **FirmwareCrashLogPayloadAddr,
+  IN     UINT32                                         EntrySize,
+  IN     EFI_GUID                                       RecordIdGuid
```

Comment:
Make sure function parameters are aligned with function comments.

Reply:
Parameters and function comments updated.